proposals/proposalTypes/Proposal.sol

```diff
@@ -0,0 +1,94 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import {Test} from "@forge-std/Test.sol";
+import {IProposal} from "@proposals/proposalTypes/IProposal.sol";
+
+import {Core} from "@protocol/core/Core.sol";
+import {Roles} from "@protocol/core/Roles.sol";
+import {Addresses} from "@proposals/Addresses.sol";
+
+abstract contract Proposal is IProposal, Test {
+    bool public DEBUG = true;
+
+    Core _core;
+
+    function setDebug(bool value) external {
+        DEBUG = value;
+    }
+
+    /// @notice run the deployment for testing only. ie intergration testing with foundry
+    /// @dev this is not run on-chain, and is only used with the foundry `forge test` command
+    function deployForTestingOnly(Addresses addresses, address deployer) public {
+        vm.startBroadcast(deployer);
```

Comment:
if you're only testing, and not running on a local anvil fork, you shouldn't need to broadcast

Reply:
This is needed for testing due to the way foundry handles the integration test deployment addresses. Without it this will fail as the deployment addresses will be the contract `TestProposals.sol` address not a set address. This approach is far more deterministic